Source/Lib/Codec/EbEntropyCoding.c

```diff
@@ -5343,6 +5343,28 @@ static void CodeProfileTierLevel(
 	return;
 }
 
+/**************************************************
+* ComputeNumReorderPics
+**************************************************/
+EB_U32 ComputeNumReorderPics(SequenceControlSet_t    *scsPtr)
+{
+    // NumReorderPics indicates the maximum number of pictures that can precede any picture in decoding 
+    // order and follow it in output order.  The value needs to be no greater than the DPB size
+    //
+    // The reorderPicsPerHierarchy values are determined by inspection of the graphs at the top of 
+    // shown at the top of EbPredictionStructure.c and determining the highest observed value for 
+    // NumReorderPics in any one picture within that prediction structure.
+    //                    hierarchical level    1  2  3  4  5   6
+    static EB_U32 reorderPicsPerHierarchy[] = { 0, 1, 2, 4, 6, 14 };
+
+    if (scsPtr->predStructPtr->predType == EB_PRED_LOW_DELAY_P || scsPtr->predStructPtr->predType == EB_PRED_LOW_DELAY_B)
+        // For low delay prediction structures set num reorder pictures to 0
+        return 0;
+    else
+        // For Random Access prediction structures set num reorder pictures according to the number of hierarchical layers
+        return MIN(scsPtr->maxDpbSize - 1, reorderPicsPerHierarchy[scsPtr->maxTemporalLayers]);
```

Comment:
no need to use MIN() here.
And your thought is for safety reason, it should be `MIN(scsPtr->maxDpbSize, reorderPicsPerHierarchy[scsPtr->maxTemporalLayers]);`, since here it is `vps_max_num_reorder_pics` not `vps_max_num_reorder_pics_minus1`

Reply:
The function calculates the value of max_num_reorder_pics (both for sps_max_num_reorder_pics and vps_max_num_reoder_pics.  Per the spec:

sps_max_num_reorder_pics[ i ] indicates the maximum allowed number of pictures with PicOutputFlag equal to 1 that can precede any picture with PicOutputFlag equal to 1 in the CVS in decoding order and follow that picture with PicOutputFlag equal to 1 in output order when HighestTid is equal to i. The value of sps_max_num_reorder_pics[ i ] shall be in the range of 0 to sps_max_dec_pic_buffering_minus1[ i ], inclusive. 

The value we are writing to the bit stream for sps_max_dec_pic_buffer_minus1 is scsPtr->maxDpbSize - 1, hence the check for in the MIN statement to make sure the value doesn't exceed  scsPtr->maxDpbSize-1 .

Reply:
I see, thx. But I think MIN is not needed.